apps/managedidentity/managedidentity_test.go

```diff
@@ -4,43 +4,401 @@ package managedidentity
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
 	"testing"
+	"time"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/errors"
+	internalTime "github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/json/types/time"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/oauth/ops/accesstokens"
+)
+
+const (
+	// test Resources
+	resource              = "https://demo.azure.com"
+	resourceDefaultSuffix = "https://demo.azure.com/.default"
 )
 
-func fakeClient(mangedIdentityId ID, options ...Option) (Client, error) {
-	client, err := New(mangedIdentityId, options...)
+type HttpRequest struct {
+	Source   Source
+	Resource string
+	Identity ID
+}
+
+type SuccessfulResponse struct {
+	AccessToken string `json:"access_token"`
+	ExpiresOn   int64  `json:"expires_on"`
+	Resource    string `json:"resource"`
+	TokenType   string `json:"token_type"`
+	ClientID    string `json:"client_id"`
+}
+
+type ErrorResponse struct {
+	StatusCode    int    `json:"statusCode"`
+	Message       string `json:"message"`
+	CorrelationID string `json:"correlationId,omitempty"`
+}
+
+type fakeClient struct{}
+type errorClient struct {
```

Comment:
🤔 you could have just one fake client type. In fact, I imagine you could use [this one](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/main/apps/internal/mock/mock.go#L49-L50) if you add a `WithStatusCode` option for it